packages/snippet-bot/test/snippet-bot.ts

```diff
@@ -580,6 +580,61 @@ describe('snippet-bot', () => {
       diffRequests.done();
     });
 
+    it('does not give warnings about wrong product prefix with snippet-bot:no-prefix-req label', async () => {
+      sandbox.restore();
+      getSnippetsStub = sandbox.stub(snippetsModule, 'getSnippets');
+      const snippets = require(resolve(fixturesPath, './snippets'));
+      getSnippetsStub.resolves(snippets);
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const diffResponse = fs.readFileSync(resolve(fixturesPath, 'diff.txt'));
+      const payload = require(resolve(
+        fixturesPath,
+        './pr_event_no_prefix_req'
+      ));
+      const blob = require(resolve(fixturesPath, './failure_blob'));
+
+      const requests = nock('https://api.github.com')
+        .get(
+          '/repos/tmatsuo/repo-automation-bots/contents/.github%2Fsnippet-bot.yml'
+        )
+        .reply(200, config)
+        .get(
+          '/repos/tmatsuo/repo-automation-bots/contents/test.py?ref=ce03c1b7977aadefb5f6afc09901f106ee6ece6a'
+        )
+        .reply(200, blob)
+        .post('/repos/tmatsuo/repo-automation-bots/check-runs', body => {
+          snapshot(body);
+          return true;
+        })
+        .reply(200)
+        .get(
+          '/repos/tmatsuo/repo-automation-bots/issues/14/comments?per_page=50'
+        )
+        .reply(200, [])
+        .post(
+          '/repos/tmatsuo/repo-automation-bots/issues/14/comments',
+          body => {
+            snapshot(body);
+            return true;
+          }
+        )
+        .reply(200);
+
+      const diffRequests = nock('https://github.com')
+        .get('/tmatsuo/repo-automation-bots/pull/14.diff')
+        .reply(200, diffResponse);
+
+      await probot.receive({
+        name: 'pull_request',
+        payload,
+        id: 'abc123',
+      });
+
+      sinon.assert.calledOnce(getSnippetsStub);
+      requests.done();
+      diffRequests.done();
+    });
+
```

Comment:
It would be good to have a test that demonstrates a status check being set to `failure`, and one setting it to success -- I'm a bit surprised we didn't need to update existing tests?

Reply:
Ok, I can add some tests with `alwaysCreateStatusCheck` config.

Reply:
Because we used `try` - `catch`, it swallowed the nock errors.

Now I added a new test case for 4 success checks.